### ISSUE TITLE
align_double_arrows and UTF8 characters

### ISSUE THREAD

**Issue by fisharebest** [FriendsOfPHP/PHP-CS-Fixer#1293]
I have an array containing UTF8 encoded characters

```php
	protected $inflect_male = array(
		'aitė\b' => 'as',
		'ytė\b'  => 'is',
		'iūtė\b' => 'ius',
		'utė\b'  => 'us',
	);
```

The `align_double_arrows` fixer converts this to

```php
	protected $inflect_male = array(
		'aitė\b'  => 'as',
		'ytė\b'   => 'is',
		'iūtė\b' => 'ius',
		'utė\b'   => 'us',
	);
```

It would appear that the fixer is counting bytes, rather than characters.

```
$ php-cs-fixer
PHP CS Fixer version 1.9.1 by Fabien Potencier (f2c2c55)
```

**Comment by keradus** [comment]
Well... I think we would need mbstring extension here. Any ideas how to fix it without that extension ?

**Comment by GrahamCampbell** [comment]
We could just use the mb_string function if it's available.

**Comment by GrahamCampbell** [comment]
Maybe we should just require the extension in 2.0?

**Comment by SpacePossum** [comment]
>Well... I think we would need mbstring extension here. Any ideas how to fix it without that extension ?

You don't want to go there ;) on version 1.x we could add `mb_string` as suggestion to install in `composer.json` and do checks during runtime to see if it is installed;

>Maybe we should just require the extension in 2.0?

:+1:

**Comment by keradus** [comment]
It is non-default extension and I woudn't want to require it.
If we want to use that ext in this method then it would be great to only suggest the extension and test it's presence in that one fixer in `supports` or somewhere like that.
What do you think ?

**Comment by fisharebest** [comment]
> Any ideas how to fix it without that extension ?

A "quick and dirty" answer is to ignore characters that match (binary) 10xxxxxx.

These are the "continuation bytes" in [UTF8](https://en.wikipedia.org/wiki/UTF-8) encoding.

**Comment by gharlan** [comment]
Or `strlen(utf8_decode())`.
But I'm not sure how reliable it is.

**Comment by keradus** [comment]
Does http://php.net/manual/pl/intro.xml.php extension is always enabled ?

**Comment by keradus** [comment]
Yeah, it is :D
http://php.net/manual/en/xml.installation.php

**Comment by keradus** [comment]
Fixed in #1313

**Comment by keradus** [comment]
Closing due to merge of #1313
